### PR TITLE
Added ChangePrice endpoint

### PR DESCRIPTION
**Pull request by epadams**
- Added ChangePrice endpoint (I know this is inconsistent naming, just following the display for the api call in the assignment)

### CODE REVIEW

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs [hunk 1]
```diff
@@ -120,5 +120,26 @@ public async Task<IActionResult> DeleteProduct(int id)
...
+        /* TODO(epadams)
+        [HttpPut("Item/UpdateStock")]
+        public async Task<IActionResult> UpdateStock(int id, int quantity)
+        {
+
+            return NoContent();
+        }
+        */
+
+        [HttpPut("Item/ChangePrice")]
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Spec says `Inventory/ChangePrice`

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs [hunk 2]
```diff
@@ -120,5 +120,26 @@ public async Task<IActionResult> DeleteProduct(int id)
...
+        public async Task<IActionResult> ChangePrice(int id, decimal price)
+        {
+            if (price < 0)
+            {
+                return BadRequest("Price must be 0 or greater");
+            }
+
+            await _productService.UpdatePrice(id, price);
+            return Ok("Price updated to " + price);
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Return the updated object, or no content.

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs [hunk 3]
```diff
@@ -120,5 +120,26 @@ public async Task<IActionResult> DeleteProduct(int id)
...
+        [HttpPut("Item/UpdateStock")]
+        public async Task<IActionResult> UpdateStock(int id, int quantity)
+        {
+
+            return NoContent();
+        }
+        */
+
+        [HttpPut("Item/ChangePrice")]
+        public async Task<IActionResult> ChangePrice(int id, decimal price)
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Add the Id to the route. The spec doesn't specify and that's what we did for the other routes.

Comment (jbytes1027) [inline ShopAPI/ShopAPI/Controllers/ItemsController.cs]: price should be in a json object to be consistent with the other endpoints. E.g. `{ "price": 124.21 }`.

**Review by jbytes1027 — CHANGES_REQUESTED** [review]
See comments.

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs (hunk 1 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Done

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs (hunk 2 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Done

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs (hunk 3 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Done

Comment (epadams) [reply ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Done

**Comment by epadams** [comment]
One bug that still needs to be fixed before merge: when calling `/api/GetCart/{id}`, it returns the stock alongside quantity in the cart. This is because the ModelToDTO for CartItems uses the ProductDTO as the base

File: ShopAPI/ShopAPI/Controllers/ItemsController.cs [hunk 4]
```diff
@@ -120,5 +120,26 @@ public async Task<IActionResult> DeleteProduct(int id)
 
             return NoContent();
         }
+
+        /* TODO(epadams)
+        [HttpPut("Item/UpdateStock")]
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Controllers/ItemsController.cs]: Don't commit/merge commented out code.

**Review by jbytes1027 — APPROVED** [review]

**Comment by jbytes1027** [comment]
> One bug that still needs to be fixed before merge: when calling `/api/GetCart/{id}`, it returns the stock alongside quantity in the cart. This is because the ModelToDTO for CartItems uses the ProductDTO as the base

What's that got to do with the ChangePrice? Create a new issue and/or pr.

**Pull request merged**